vault/expiration.go

```diff
@@ -114,6 +116,58 @@ func (c *Core) stopExpiration() error {
 	return nil
 }
 
+// Tidy cleans up stale leases which are associated with invalid tokens
+func (m *ExpirationManager) Tidy() error {
+	var tidyErrors *multierror.Error
+
+	tidyFunc := func(leaseID string) {
+		le, err := m.loadEntry(leaseID)
+		if err != nil {
```

Comment:
I think it's worth adding a memoizing cache (a map would work, as a set) so that if we already know a token *does* still exist we don't have to look it up again.